components/ecodan/response.cpp

```diff
@@ -186,7 +186,13 @@ namespace ecodan
                 publish_state("status_water_pump_2", status.WaterPump2Active);
                 publish_state("status_three_way_valve_2", status.ThreeWayValve2Active);                
                 //ESP_LOGI(TAG, res.debug_dump_packet().c_str());
-                break;                
+                break;          
+            case GetType::ADDITIONAL_PUMP_STATUS:
+                // byte 2 = water pump 3 (out3 - Zone2) running on/off       
+                status.WaterPump3Active = res[2] != 0 || true;

```

Comment:
curious, why the || true ?

Reply:
That was just a test to see if that part of code was executed or not. (I mentioned it in https://github.com/gekkekoe/esphome-ecodan-hp/pull/73#discussion_r1852601160)

Reply:
ah good, lets wait for 1 additional user report, and then I will merge it

Reply:
Not yet - now I am fetching both x15 & x16 responses and:
![image](https://github.com/user-attachments/assets/bedb4f5e-bb93-4f6c-bca8-3f356abe97d2)
So it looks like that I get WaterPump3 status in byte 5 of x15 response - but I need to wait few hours to get to DHW cycle to confirm this.

Reply:
you can increase debug level in yaml and read the logged msg :)

hmm byte 5 of 0x15. 0x16 is not reporting anything for you. @F1p 0x16 worked for you right?

Reply:
Procon confirms 0x15 bytes 4/5/6/7 matches Discrete Input (Digital Input):
![image](https://github.com/user-attachments/assets/69c94f33-c723-403f-bd56-87036f190aeb)

I got strange results from holding registers though.

"Water Pump 1/2/3" may not link or match FTC board marking "OUT" 1/2/3 though...


I worked with a FTC6 user extensively on pumps and we looked at PCB outputs and linked them to bits.
My Zone2 pump and "status of heating" align with 0x16 byte 2 and the pump is connected to PCB marking "OUT3"
My DHW water pump is connected to CPN4 and aligns with 0x16 byte 1
![image](https://github.com/user-attachments/assets/e71d19a4-1c45-48ae-9fb7-43cc1a402c24)

Reply:
> Not yet - now I am fetching both x15 & x16 responses and: ![image](https://private-user-images.githubusercontent.com/44526436/389176165-bedb4f5e-bb93-4f6c-bca8-3f356abe97d2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIzNTgyMjUsIm5iZiI6MTczMjM1NzkyNSwicGF0aCI6Ii80NDUyNjQzNi8zODkxNzYxNjUtYmVkYjRmNWUtYmI5My00ZjZjLWJjYTgtM2YzNTZhYmU5N2QyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIzVDEwMzIwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4ZGJhMWRlMjRjNWYyYmE3YmIxYTMxOTk1NDNjMTY1Y2QxNGE2NmQwNDAyOWIwNmViZGYyOGY0YTM2ZDRkMWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6lU7oky4QSBjdcyGmi2aR6bstUsh5nCj-UxKMRkYv54) So it looks like that I get WaterPump3 status in byte 5 of x15 response - but I need to wait few hours to get to DHW cycle to confirm this.

Can you trace the cable to see what "OUT" marking on the PCB is triggering for this byte please?

Reply:
@DamijanD So per @F1p byte 5 is pump 3. Can you see where it's connected on your main board ? Is it connected to one of the OUT 1/2/3 sockets or something else ?

Reply:
I analysed the 0x15 response and found out this:
- Byte 5 - water pump 3 (out3) ON/OFF
- Byte 11 - Mixing valve  (out5) : 0:idle 1:closing 2:opening 3:close/stop
- Byte 10 - this has maybe something to do with mixing valve position (values 0, 8, 9, A) 

@F1p:
![image](https://github.com/user-attachments/assets/bfe024a3-69c7-46c3-a112-4e64d8e0dc6b)
![image](https://github.com/user-attachments/assets/99a544ee-5c42-48e2-9ea0-dbcdfc07c047)

Reply:
I asked @F1p to check it with the procon since I don't have a procon at my disposal. byte 5 is correct.
We can remove 0x16 from the query then, and update protocol.md.

If you also want the mixing valve status, that one can be added as well. Probably best to keep it numeric and translate it in the yaml.

btw, when you have some time, also have a look at: https://github.com/gekkekoe/esphome-ecodan-hp/discussions/70#discussioncomment-11349614

Reply:
Going to monitor my 0x15 byte 5 to compare with 0x16 byte 3


![image](https://github.com/user-attachments/assets/884a96a6-3a39-43a7-97a2-2ad594e6f2d9)

Reply:
Byte 3 is not static for me: (most of the time value is 1F)
RES{.Hdr{FC,62,02,7A,10}.Payload{15,01,1F,1F,01,01,00,01,00,00,0A,00,00,00,00,00}.Chk{B1}}
RES{.Hdr{FC,62,02,7A,10}.Payload{15,01,1F,20,01,01,00,01,00,00,09,00,00,00,00,00}.Chk{B1}}
RES{.Hdr{FC,62,02,7A,10}.Payload{15,01,1F,1E,00,00,01,00,01,01,00,03,00,00,00,00}.Chk{B9}}
RES{.Hdr{FC,62,02,7A,10}.Payload{15,01,1F,1D,00,00,01,00,00,00,00,00,00,00,00,00}.Chk{BF}}

@F1p Do you also have mixing valve for Zone2? I don't know how it can be set but in melcloud response there is `"HasSimplifiedZone2": false,` so maybe 0x15 is used for "proper" Zone2 and 0x16 for "simplified"  ?

Reply:
Never had MELCloud, but no mixing tank here so perhaps i do have simplified zone2..

Reply:
Probably mixing tank is not what is important here but motorized mixing valve...